### DISCUSSION TITLE
Processing EquCab3 files

### DISCUSSION THREAD

**Discussion by prmunn** [GreenleafLab/ArchR#1393] [Questions / Documentation]
Hi - I'm attempting to process some fragment files generated from EquCab3 (horse) samples, but this genome doesn't appear to be built in to ArchR. I looked into the createGenomeAnnotations function, but that uses genomes that have been set up in Bioconductor's BSgenomes and they don't have EquCab set up either. Is there another way to set up (or import) genome annotation files?

**Comment by rcorces** [comment]
ArchR does require a `BSgenome` object. If your species of interest doesnt have one, you would have to forge it yourself. That process is outlined here - https://bioconductor.org/packages/devel/bioc/vignettes/BSgenome/inst/doc/BSgenomeForge.pdf

You should be sure that your forged BSgenome matches exactly the reference sequence used for alignment.

**Reply by prmunn** [reply]
Ok - I'll give that a try. Thanks

**Reply by prmunn** [reply]
I appear to be missing a critical step. I've created the the BSgenome.Equuscabellus.UCSC.equCab3_1.0.0.tar.gz file by going through the instructions in BSgenomeForge.pdf, but it's not clear to me how I use this file from within Rstudio. Specifically, I'm trying to use createGenomeAnnotation(genome="BSgenome.Equuscabellus.UCSC.equCab3") but I'm not sure how Rstudio can access the forged file.

**Reply by rcorces** [reply]
How far did you go in the process? Did you do this:

```
Once forgeBSgenomeDataPkg is done, ignore the warnings (if any), quit R, and build the source package
(tarball) with
R CMD build <pkgdir>
where <pkgdir> is the path to the source tree of the package.
Then check the package with
R CMD check <tarball>
where <tarball> is the path to the tarball produced by R CMD build.
Finally install the package with
R CMD INSTALL <tarball>
and use it!
```

If you've installed the package, then I think you would run `library(BSgenome.Equuscabellus.UCSC.equCab3_1.0.0)` or similar to load the BSgenome object into R.

**Reply by prmunn** [reply]
That was the step I was missing. I wasn't sure how to load the object into R. Thanks!

**Reply by prmunn** [reply]
Next issue. I was able to load the object into R and ran the "createGenomeAnnotation" function. However, when I attempt to run the arrArchRGenome function I get the following message:

Genome : BSgenome.Equuscabellus.UCSC.equCab3 is not currently supported by ArchR.
Currently supported genomes : hg19,hg38,mm9,mm10,hg19test
To continue try building a custom geneAnnotation with createGeneAnnnotation(),
and genomeAnnotation with createGenomeAnnotation()!

The help for addArchRGenome suggested that I need to specift the genome for each function I run (using the genomeAnnotation parameter). Is there a way I can define it globally for all functions?

**Reply by rcorces** [reply]
Once you create an ArchRProject, the gene and genome annotations will be stored there. You cannot use `addArchRGenome()` with a custom genome.

**Reply by prmunn** [reply]
Ok. Thanks!
